src/layers/XRLayer/XRLayer.js

```diff
@@ -150,35 +164,48 @@ export default class XRLayer extends Layer {
       channel2: null,
       channel3: null,
       channel4: null,
-      channel5: null
+      channel5: null,
+      channelColormap: null
     };
     if (this.state.textures) {
       Object.values(this.state.textures).forEach(tex => tex && tex.delete());
     }
     if (channelData.length > 0) {
-      channelData.forEach(function(d, i) {
-        textures[`channel${i}`] = this.dataToTexture(d);
-      }, this);
-      this.setState({ textures });
+      if (this.props.colormap) {
+        // assume first dimension is the one we want to fetch
+        textures.channelColormap = this.dataToTexture(channelData[0]);
+        this.setState({ textures });
+      } else {
+        channelData.forEach(function(d, i) {
+          textures[`channel${i}`] = this.dataToTexture(d);
+        }, this);
+        this.setState({ textures });
+      }
     }
   }
 
   dataToTexture(data) {
-    const isInt8 = data instanceof Uint8Array;
-    const isInt16 = data instanceof Uint16Array;
-    const isInt32 = data instanceof Uint32Array;
+    const { dtype } = this.props;
```

Comment:
I think it would be good to extract this into a hash table somehow.

```javascript
const GL_FORMAT_DTYPE = {
     "<u1": { format: GL.R8UI, type: GL.UNSIGNED_BYTE }, 
     "<u2": { format: GL.R16UI, type: GL.UNSIGNED_SHORT }, 
     "<u4": { format: GL.R32UI, type: GL.UNSIGNED_INT }, 
     "<f4": { format: GL.R32F, type: GL.FLOAT }, 
}
```

Reply:
I agree.